pgtransaction/transaction.py

```diff
@@ -108,6 +108,68 @@ def atomic(
     isolation_level=None,
     retry=None,
 ):
+    """
+    Extends django.db.transaction.atomic with PostgreSQL functionality.
+
+    An extension of the django.db.transaction.atomic function
+    which allows one to dynamically set the isolation level when opening a transaction,
+    as well as specifying a retry policy for when an operation in that transaction results
+    in a Postgres locking exception.
+
+    Usage:
+        Since ``pgtransaction.atomic`` inherits from ``django.db.transaction.atomic``, it
+        can be used in exactly the same manner. Additionally, when used as a
+        context manager or a decorator, one can use it to specify the
+        isolation level of the new transaction. For example:
+
+        .. code-block:: python
+
+            with pgtransaction.atomic(isolation_level=pgtransaction.REPEATABLE_READ):
+                # Isolation level is now REPEATABLE READ for the duration of the "with" block.
+                ...
+
+        Note that setting isolation_level in a nested atomic block is not permitted,
+        and attempting to do so will result in a RuntimeError.
+
+        When used as a decorator, one can also specify a ``retry`` argument. This
+        defines the number of times the transaction will be retried upon encountering
+        the exceptions referenced by ``settings.PGTRANSACTION_RETRY_EXCEPTIONS``,
+        which defaults to ``(psycopg2.errors.SerializationFailure, psycopg2.errors.DeadlockDetected)``.
+        For example:
+
+        .. code-block:: python
+
+            @pgtransaction.atomic(retry=3)
+            def update():
+                # will retry update function up to 3 times
+                # whenever any exception in settings.PGTRANSACTION_RETRY_EXCEPTIONS
+                # is encountered. Each retry will open a new transaction (after
+                # rollback the previous one).
+
+        Attempting to set a non-zero value for ``retry`` when using ``pgtransaction.atomic``
+        as a context manager will result in a RuntimeError.
+
+    Args:
+        isolation_level (str): The isolation level we wish to be
+            used for the duration of the transaction. If passed in
+            as None, the current isolation level is used. Otherwise,
+            we must choose from ``pgtransaction.READ_COMMITTED``,
+            ``pgtransaction.REPEATABLE_READ`` or ``pgtransaction.SERIALIZABLE``.
+            Note that the default isolation for a Django project is
+            "READ COMMITTED". It is not permitted to pass this value
+            as anything but None when using ``pgtransaction.atomic``
+            is used as a nested atomic block - in that scenario,
+            the isolation level is inherited from the parent transaction.
```

Comment:
This last sentence is no longer true after the updates. One can do the following successfully:

```
with pgtransaction.atomic():
    with pgtransaction.atomic(isolation_level=pgtransaction.SERIALIZABLE):
        # And so on...
```

It can be nested as many times until the first statement executes. Then it will error. Note that you can still use the same isolation level even after executing a statement:

```
with pgtransaction.atomic(isolation_level=pgtransaction.SERIALIZABLE):
    # Do a statement that queries the DB...
    User.objects.get(...)
    # You can still use the same isolation level
    with pgtransaction.atomic(isolation_level=pgtransaction.SERIALIZABLE):
        # This will fail
        with pgtransaction.atomic(isolation_level=pgtransaction.REPEATABLE_READ):
```